### DISCUSSION TITLE
Setting a higher Max Feedrate for Retraction

### DISCUSSION THREAD

**Discussion by Xploder** [Jyers/Marlin#314] [Ideas]
Hey guys and girls,

so in pretty much every default Ender3V2 config the max feedrate for the extruder is set to 25mm/s, while most Slicer configs use something like 45mm/s for retraction. Marlin will always auto-correct these retractions down to 25mm/s.

What do you think? Should we raise the Max Feedrate in the config for all print moves or maybe even investigate the [FWRETRACT](https://marlinfw.org/docs/features/fwretract.html) feature?
This would have the advantage of [using a different feedrate setting just for retraction](https://marlinfw.org/docs/gcode/M207.html) which can be a lot higher than the max print feedrate while having the safety of never going over 25mm/s during "real" extrudes.

Looking forward to your experiences with this :)

**Comment by Schnabulation** [comment]
I never really got my head around the retraction feedrate. I'm using Cura with the default profile where retraction is set to 25 mm/s and I was quite happy with this so far. What would be the benefit of retracting faster?

**Reply by Xploder** [reply]
A higher retraction speed can help make your retraction better because it removes the filament from the nozzle faster and therefore reduces oozing.
It will also speed up your print especially on gcode that has many retractions. On a project I'm currently slicing it's 2h33m vs. 2h43m at 40mm/s compared to 25mm/s.

I've tuned mine to 40mm/s retraction on my BMG clone extruder now and will keep tuning it.

**Reply by DerPit** [reply]
Hmm, maybe that explains why I couldn't improve on oozing by increasing the retraction speed?
Is that limit on purpose, i.e.,  is the (original) extruder stepper of the E3v2 overheating with higher speeds?

**Comment by joeclifford** [comment]
Jyers config for max feedrate matches the main marlin branch for the Ender 3 V2 as well as Creality's own source files

https://github.com/MarlinFirmware/Configurations/blob/import-2.0.x/config/examples/Creality/Ender-3%20V2/Configuration.h#L808

https://file2-cdn.creality.com//website/f120fc43-87a3-4f5e-87d4-0cb3cd0945c0

I think investigating firmware retract sound like a much better idea than changing the defaults away from the upstream branches.

Users can always set the feedrate themselves using gcode M203 if they want to.

https://marlinfw.org/docs/gcode/M203.html
